std/array.d

```diff
@@ -1953,7 +1953,7 @@ private enum bool hasCheapIteration(R) = isArray!R;
    See_Also:
         For a lazy version, see $(REF joiner, std,algorithm,iteration)
   +/
-ElementEncodingType!(ElementType!RoR)[] join(RoR, R)(RoR ror, scope R sep)
+ElementEncodingType!(ElementType!RoR)[] join(RoR, R)(RoR ror, R sep)
```

Comment:
this removes `scope` I thought this PR was trying to add it, am I missing something?

Reply:
Range functions should ideally infer `scope` based on `empty()`, `popFront()`, `front()`, copy constructors etc. 
`scope` inference is not that good yet however ([issue 20674](https://issues.dlang.org/show_bug.cgi?id=20674)) so sometimes `scope` is manually applied to help the compiler. With issue 20150 fixed, this `scope R sep` gives an error:

> scope variable `sep` assigned to non-scope parameter `r` calling std.array.array!(Once).array

So the long term fix is to ensure `scope` inference works for both `join` and `array`, but that's an enhancement. For now it's important to get the dip1000 bug fix in.

Reply:
Won't this break existing code if the inference fails?

Reply:
All the accepts-invalid fixes for dip1000 are breaking changes. Luckily dip1000 is still experimental.

Reply:
But these changes are neither specific to DIP1000 nor about actually invalid code. It would be better to first improve the inference s.t. there is no unecessary breakage.

Reply:
What code do you think would break without `-preview=dip1000`?

Reply:
I don't have aspecific example ATM, will post one if I find a small example. 
But we should strive to not break valid code. `-dip1000` has existed for quite some time and is certainly used in the wild.

Reply:
https://github.com/dlang/dmd/pull/12010 is going to break such dip1000 code in the wild undoubtedly, I just tested it on my own 37 KLOC dip1000 library and I had to add 3 `scope` annotations. Do you think we need a second switch `preview=dip1000fixed`?

Reply:
Neither said nor implied that - some breakage will be unavoidable. I'm just concerned that these changes introduce a temporary regression that could be avoided by first improving the `scope` inference.

But the actual fallout might be neglegible given that your projects required minimal changes and buildkite passes as is.